src/main/java/com/playonlinux/ui/api/RemoteAvailableInstallers.java

```diff
@@ -34,13 +36,82 @@
 
     boolean hasFailed();
 
-    Iterable<ScriptDTO> getAllScripts();
+    Iterable<ScriptDTO> getList();
 
-    Iterable<ScriptDTO> getAllScripts(String filter);
+    InstallerFilter getFilter();
 
-    Iterable<ScriptDTO> getAllScriptsInCategory(String categoryName) throws PlayOnLinuxError;
-
-    ScriptDTO getScriptByName(String scriptName) throws PlayOnLinuxError;
+    ScriptDTO getByName(String scriptName) throws PlayOnLinuxError;
 
     void refresh();
+
+
+    /**
+     * This class is handling the filtering of Installers, providing one central place for
+     * changing how the filtering is done.
+     * InstallerFilters extends Observable, so that changes get directly promoted and
+     * initiate a new filtering process.
+     */
+    abstract class InstallerFilter extends Observable implements Filter<ScriptDTO> {
+
+        protected String title;
+        protected String category;
+        protected boolean showTesting;
+        protected boolean showNoCd;
+        protected boolean showCommercial;
+
+        public String getTitle() {
+            return title;
```

Comment:
This abstract class is needed within RemoteAvailableInstallers, so that we can overload `Filterable<DTO>`'s getFilter(), with the more specific filter for this DTO-Container. Without this, all users of the `getFilter()` function would have to cast the returned `Filter<DTO>` to the more concrete `InstallerFilter` to be able to set filter-parameters.